DDOT_module/child_classes/correction_data_generator.py

```diff
@@ -0,0 +1,397 @@
+"""
+Created on 29.04.2021
+
+@author: Filip Maciejewski
+@contact: filip.b.maciejewski@gmail.com
+"""
+
+import copy
+import numpy as np
+from typing import Optional, Dict
+from tqdm import tqdm
+from QREM import ancillary_functions as anf
+from ..child_classes.ddot_marginal_analyzer_vanilla import DDOTMarginalsAnalyzer
+from DDOT_module.child_classes.global_noise_matrix_creator import GlobalNoiseMatrixCreator
+
+
+class CorrectionDataGenerator(DDOTMarginalsAnalyzer):
+    """
+        1
+    """
+
+    def __init__(self,
+                 results_dictionary_ddot: dict,
+                 reverse_counts: bool,
+                 number_of_qubits: int,
+                 marginals_dictionary: dict,
+                 clusters_list: list,
+                 neighborhoods: dict,
+                 noise_matrices_dictionary: Optional[dict] = None
+                 ) -> None:
+
+        super().__init__(results_dictionary_ddot,
+                         reverse_counts,
+                         marginals_dictionary,
+                         noise_matrices_dictionary
+                         )
+        self._number_of_qubits = number_of_qubits
+        self._qubit_indices = range(number_of_qubits)
+
+        self._clusters_list = clusters_list
+        self._neighborhoods = neighborhoods
+
+        clusters_dictionary = {'q%s' % qi: [] for qi in self._qubit_indices}
+        for cluster in clusters_list:
+            for qi in cluster:
+                for qj in cluster:
+                    if qi != qj:
+                        clusters_dictionary['q%s' % qi].append(qj)
+
+        self._clusters_dictionary = clusters_dictionary
+
+        self._noise_matrices = {}
+        self._correction_matrices = {}
+        self._mitigation_errors = {}
+        self._correction_indices = {}
+
+    def compute_pairs_correction_matrices(self,
```

Comment:
This method should absolutely be decomposed into several smaller ones. It's too long.

Reply:
I agree, this method is "the worst" of all methods in this repository as of now. :D I'll think about it.